src/models.rs

```diff
@@ -0,0 +1,38 @@
+use near_sdk::serde::{Serialize, Deserialize};
+
+use crate::*;
+
+/// Information about a specific public key. Should be returned in the `get_key_information` view method.
+/// Part of the linkdrop NEP
+#[derive(Serialize)]
+#[serde(crate = "near_sdk::serde")]
+pub struct KeyInfo {
+    /// yoctoNEAR$ amount that will be sent to the claiming account (either new or existing)
+    /// when the key is successfully used.
+    pub balance: U128,
+}
+
+
+#[derive(Serialize, Deserialize)]
+#[serde(crate = "near_sdk::serde")]
+/// Information about any limited access keys that are being added to the account as part of `create_account_advanced`.
+pub struct LimitedAccessKey {
+    /// The public key of the limited access key.
+    pub public_key: PublicKey,
+    /// The amount of yoctoNEAR$ that can be spent on Gas by this key.
+    pub allowance: U128,
+    /// Which contract should this key be allowed to call.
+    pub receiver_id: AccountId,
+    /// Which methods should this key be allowed to call.
+    pub method_names: String,
```

Comment:
Maybe it would be better to make this an Array of strings, and then stringify it when using it. This would be less error prone for the users.

Reply:
The reason behind keeping it a string is that's what's done on the protocol side and is pretty standard for limited access keys from a contract perspective.